### PR TITLE
ADD and CHANGE_DATA will now be automatically decided

### PR DESCRIPTION
**Pull request by pawbu**

### CODE REVIEW

File: datastore_version_manager/adapter/datastore.py [hunk 1]
```diff
@@ -227,6 +227,10 @@ def write_metadata_all(metadata_all: dict, version: str) -> None:
 
 
 def is_dataset_in_datastore_versions(dataset_name: str, release_status) -> bool:
+    """
+    Determines if dataset has existed in the datastore.
+    :return: This function returns True also in case dataset was RELEASED and then DELETED.
+    """
```
Comment (DanielElisenberg) [inline datastore_version_manager/adapter/datastore.py:233]: :100:

File: datastore_version_manager/api/command_api.py [hunk 2]
```diff
@@ -30,17 +30,15 @@ def get_pending_operations():
 def add_pending_operation(body: NewPendingOperationRequest):
     logger.info(f'POST /pending-operations with body {body}')
     operation_type = body.operationType
-    if operation_type == 'ADD':
+    if operation_type == 'ADD_OR_CHANGE_DATA':
```
Comment (DanielElisenberg) [inline datastore_version_manager/api/command_api.py:33]: can we call this IMPORT_DATA or similar?

File: datastore_version_manager/domain/draft_dataset.py [hunk 3]
```diff
@@ -13,22 +13,26 @@
...
-    if operation_type == 'ADD':
-        if datastore.is_dataset_in_datastore_versions(dataset_name, "RELEASED"):
+                          dataset_name: str, description: str) -> None:
+    if operation_type == 'ADD_OR_CHANGE_DATA':
+        if datastore.is_dataset_in_datastore_versions(dataset_name, "DELETED"):
             raise ForbiddenOperation(
-                f'Can not add new variable "{dataset_name}". '
-                'A versioned variable of the same name already exists in datastore'
+                f'Cannot add or update variable "{dataset_name}". '
+                'This variable has been removed from datastore'
```
Comment (DanielElisenberg) [inline datastore_version_manager/domain/draft_dataset.py:21]: :100:

**Review by DanielElisenberg — APPROVED** [review]
This looks wonderful :100: great error messages and comments 🚀 
One minor naming comment

File: datastore_version_manager/api/command_api.py (hunk 2 above)
Comment (pawbu) [reply datastore_version_manager/api/command_api.py:33]: I would leave it like that as it points to the concepts that we have used so far and are familiar with.

**Pull request merged**